packages/template-retail-react-app/app/hooks/use-product-view-modal.js

```diff
@@ -5,30 +5,54 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {useEffect, useState} from 'react'
+import {useEffect} from 'react'
 import {rebuildPathWithParams, removeQueryParamsFromPath} from '../utils/url'
 import {useHistory, useLocation} from 'react-router-dom'
-import {useCommerceAPI} from '../commerce-api/contexts'
-import {isError} from '../commerce-api/utils'
 import {useVariant} from './use-variant'
 import {useToast} from './use-toast'
 import {useIntl} from 'react-intl'
 import {API_ERROR_MESSAGE} from '../constants'
+import {useProduct} from 'commerce-sdk-react-preview'
 
 /**
- * This hooks is responsible for fetching a product detail based on the variation selection
+ * This hook is responsible for fetching a product detail based on the variation selection
  * and managing the variation params on the url when the modal is open/close
  * @param initialProduct - the initial product when the modal is first open
  * @returns object
  */
 export const useProductViewModal = (initialProduct) => {
     const location = useLocation()
-    const api = useCommerceAPI()
     const history = useHistory()
     const intl = useIntl()
-    const [product, setProduct] = useState(initialProduct)
-    const [isFetching, setIsFetching] = useState(false)
     const toast = useToast()
+    const urlParams = new URLSearchParams(location.search)
+    const pid = urlParams.get('pid')
+    // use commerce-sdk-react-preview useProduct to fetch new product detail whenever variant changes
+    const {data: product, isFetching} = useProduct(
+        {id: pid},
+        {
+            // Show initialTodos immediately, but won't refetch until another interaction event is encountered after 1000 ms
+            initialData: initialProduct,
+            // when the modal is first mounted, don't need to fetch current product detail since it is available
+            enabled: !!pid,
+            onSuccess: (data) => {
+                console.log('data.productId', data.id)
+                // if the product id is the same as the initial product id, then merge the data with the initial product
```

Comment:
If it doesn't match, we do nothing?

Reply:
onSuccess was not accurate, I believe `select` is the one we want to use here. select allows us to transform the data before returning it